src/equations/shallow_water_2d.jl

```diff
@@ -0,0 +1,564 @@
+is useful to set initial conditions or test the "lake-at-rest" well-balancedness.
+
+The bottom topography function ``b(x,y)`` is set inside the initial condition routine
+for a particular problem setup. To test the conservative form of the SWE one can set the bottom topography
+variable `b` to zero.
+
+In addition to the unknowns, Trixi currently stores the bottom topography values at the approximation points
+despite being fixed in time. This is done for convenience of computing the bottom topography gradients
+on the fly during the approximation as well as computing auxiliary quantities like the total water height ``H``
+or the entropy variables.
+This affects the implementation and use of these equations in various ways:
+* The flux values corresponding to the bottom topography must be zero.
+* The bottom topography values must be included when defining initial conditions, boundary conditions or
+  source terms.
+* [`AnalysisCallback`](@ref) analyzes this variable.
+* Trixi's visualization tools will visualize the bottom topography by default.
+
+References for the SWE are many but a good introduction is available in Chapter 13 of the book:
+- Randall J. LeVeque (2002)
+  Finite Volume Methods for Hyperbolic Problems
+  [DOI: 10.1017/CBO9780511791253](https://doi.org/10.1017/CBO9780511791253)
+"""
+struct ShallowWaterEquations2D{RealT<:Real} <: AbstractShallowWaterEquations{2, 4}
+  gravity::RealT # gravitational constant
+  H0::RealT      # constant "lake-at-rest" total water height
+end
+
+# Allow for flexibility to set the gravitational constant within an elixir depending on the
+# application where `gravity_constant=1.0` or `gravity_constant=9.81` are common values.
+# The reference total water height H0 defaults to 0.0 but is used for the "lake-at-rest"
+# well-balancedness test cases
+function ShallowWaterEquations2D(gravity_constant; H0=0.0)
+  ShallowWaterEquations2D(gravity_constant, H0)
+end
+
+
+have_nonconservative_terms(::ShallowWaterEquations2D) = Val(true)
+varnames(::typeof(cons2cons), ::ShallowWaterEquations2D) = ("h", "h_v1", "h_v2", "b")
+varnames(::typeof(cons2prim), ::ShallowWaterEquations2D) = ("H", "v1", "v2", "b") # total water height: H = h + b
+
+
+# Set initial conditions at physical location `x` for time `t`
+"""
+    initial_condition_constant(x, t, equations::ShallowWaterEquations2D)
+
+A constant initial condition to test free-stream preservation / well-balancedness.
+"""
+function initial_condition_well_balancedness(x, t, equations::ShallowWaterEquations2D)
+  # Set the background values
+  H = equations.H0
+  v1 = 0.0
+  v2 = 0.0
+  # bottom topography taken from Pond.control in [HOHQMesh](https://github.com/trixi-framework/HOHQMesh)
+  x1, x2 = x
+  b = (  1.5 / exp( 0.5 * ((x1 - 1.0)^2 + (x2 - 1.0)^2) )
+       + 0.75 / exp( 0.5 * ((x1 + 1.0)^2 + (x2 + 1.0)^2) ) )
+  return prim2cons(SVector(H, v1, v2, b), equations)
+end
```

Comment:
Do you mean on the line or that the docstring makes no sense?

Reply:
It says `initial_condition_constant` in the docstring for a function `initial_condition_well_balancedness`

Reply:
Done. This docstring was removed because this initial condition moved into the elixir file as suggested [above](https://github.com/trixi-framework/Trixi.jl/pull/844#discussion_r711578274)